include/seastar/core/fstream.hh

```diff
@@ -55,7 +65,7 @@ class file_input_stream_history {
 struct file_input_stream_options {
     size_t buffer_size = 8192;    ///< I/O buffer size
     unsigned read_ahead = 0;      ///< Maximum number of extra read-ahead operations
-    ::seastar::io_priority_class io_priority_class = default_priority_class();
+    internal::wrapped_io_priority_class io_priority_class = {};
```

Comment:
Couldn't we just mark it as [[deprecated]]? Or will it generate deprecation warnings whenever someone instantiates file_input_stream_options?

With compile-time solution, it would just not be there.

Reply:
It will generate deprecation warning every time it's instantiated _and_ copied/moved, but the latter could (probably) be solved with un-deprecated explicit copy/move constructors and assign operators. Anyway...

Reply:
I dropped this place from the new version. Instead, the io_priority_class is ifdef-ed out for API level 7. So users that ignore API level update and only use fstream options won't notice it :( I don't think it's bad

Reply:
They'll notice deprecated create_io_priority_class(), no?

Reply:
Yes, they'll notice `register_one_priority_class` and `default_priority_class` deprecation and then would need to find where they actually use it. Not big deal either, of course